shared/js/ui/models/site.es6.js

```diff
@@ -248,7 +248,20 @@ Site.prototype = window.$.extend({},
                 this.isWhitelisted = !this.isWhitelisted
                 this.set('whitelisted', this.isWhitelisted)
                 const whitelistOnOrOff = this.isWhitelisted ? 'off' : 'on'
-                this.fetch({ firePixel: ['ept', whitelistOnOrOff] })
+
+                // fire ept.on pixel if just turned privacy protection on,
+                // fire ept.off pixel if just turned privacy protection off.
+                if (whitelistOnOrOff === 'on') {
+                    // If user reported broken site and opted to share data on site,
+                    // attach domain to ept.on pixel if they turn privacy protection back on.
+                    if (this.whitelistOptIn) {
+                        this.fetch({ firePixel: ['ept', 'on', this.tab.site.domain] })
```

Comment:
Could this whole block be simplified? it seems like there are really 2 cases here: is whitelist on + did the user opt in? ---> Then send additional data.
Otherwise, whether it's on or off ----> just send whitelistOnOrOff

Reply:
Totally. I should've mentioned that I need to revisit this block anyway, as that whitelistOptIn actually isn't hooked into anything at the moment.